MonoGame.Framework/Utilities/Png/PngWriter.cs

```diff
@@ -211,6 +211,16 @@ private void GetColorData(Texture2D texture2D)
 
             switch (texture2D.Format)
             {
+                case SurfaceFormat.Single:
+                    var floatData = new float[colorDataLength];
+                    texture2D.GetData<float>(floatData);
+                    
+                    for (int i = 0; i < colorDataLength; i++) {
+                        // Export as a greyscale image.
+                        colorData[i] = new Color(floatData[i], floatData[i], floatData[i]);
```

Comment:
My latest commit addresses the problems you've pointed out. 

I've not changed to using fixed yet, just keeping it in consistent with how arrays are iterated over in the other case blocks. When I get some more time I will do some profiling and see what kind of performance increase this would yield.